classes/gateways/class.pmprogateway_stripe.php

```diff
@@ -1692,36 +1692,45 @@ public static function pmpro_billing_preheader_stripe_checkout() {
 	 * @since 2.10.
 	 */
 	public static function pmpro_billing_preheader_stripe_customer_portal() {
-		if ( 'portal' === get_option( 'pmpro_stripe_update_billing_flow' ) ) {
-			// Get current user.
-			$user = wp_get_current_user();
-			if ( empty( $user->ID ) ) {
-				$error = __( 'User is not logged in.', 'paid-memberships-pro' );
-			}
+		global 	$pmpro_billing_order;
+		//Bail if the customer portal isn't enabled
+		if ( 'portal' !== pmpro_getOption( 'stripe_update_billing_flow' ) ) {
+			return;
+		}
 
-			if ( empty( $error ) ) {
-				// Get the Stripe Customer.
-				$stripe = new PMProGateway_stripe();
-				$customer = $stripe->get_customer_for_user( $user->ID );
-				if ( empty( $customer->id ) ) {
-					$error = __( 'Could not get Stripe customer for user.', 'paid-memberships-pro' );
-				}
-			}
+		//Bail if the order's gateway isn't Stripe
+		if ( 'stripe' !== $pmpo_billing_order->gateway  ) {
```

Comment:
```suggestion
		if ( empty( $pmpro_billing_order->gateway ) || 'stripe' !== $pmpro_billing_order->gateway  ) {
```
Fixing variable name typo and making sure that we have a billing order with a gateway set.

Reply:
good catch on the typo thanks. What's weird is i'm a hundred percent i debugged this one and i saw the global object containing the gateway 🤔  not sure when i introduced the typo bug, perhaps in the second commit. Anyway thanks for catching it.

Reply:
done.